modules/types/src/transfers.ts

```diff
@@ -4,18 +4,40 @@ import {
   HashLockTransferAppName,
   SimpleLinkedTransferAppName,
   SimpleSignedTransferAppName,
-  CoinTransfer,
+  SupportedApplications,
+  GenericConditionalTransferAppName,
 } from "./contracts";
 
 ////////////////////////////////////////
 // Types
 
-export const GenericConditionalTransferAppName = "GenericConditionalTransferApp";
+export type TransferType = "RequireOnline" | "AllowOffline";
+export const getTransferTypeFromAppName = (
+  name: SupportedApplications,
+): TransferType | undefined => {
+  switch (name) {
+    case SupportedApplications.DepositApp:
+    case SupportedApplications.SimpleTwoPartySwapApp:
+    case SupportedApplications.WithdrawApp: {
+      return undefined;
```

Comment:
It's not obvious what an undefined return value means here. If it doesn't apply, we could explicitly return a "NotApplicable" `TransferType`.
Additionally, it seems like we use this fn to tag just a few apps where it *is* relevant, therefore we should default to `undefined` for all others.

Defining exceptions out of existence is recommended by A Philosophy of Software design. The highlights I made while reading this book are available here: https://gitlab.com/bohendo/my-clippings/-/blob/master/highlights/a-philosophy-of-software-design.md
> loc 1827 - This chapter focused on exceptions, which are one of the most significant sources of special-case code, and discussed how to reduce the number of places where exceptions must be handled. The best way to do this is by redefining semantics to eliminate error conditions. For exceptions that can’t be defined away, you should look for opportunities to mask them at a low level, so their impact is limited, or aggregate several special-case handlers into a single more generic handler.